singleton/controller.go

```diff
@@ -27,6 +31,39 @@ func AsReconciler(reconciler Reconciler) reconcile.Reconciler {
 	})
 }
 
+type ChannalObjectReconciler[T client.Object] interface {
+	Reconcile(ctx context.Context, object T) (reconcile.Result, error)
+}
+
+func AsChannelObjectReconciler[T client.Object](watchEvents <-chan watch.Event, reconciler ChannalObjectReconciler[T]) reconcile.Reconciler {
+	return reconcile.Func(func(ctx context.Context, r reconcile.Request) (reconcile.Result, error) {
+		var errs error
+		var results []reconcile.Result
+		e := <-watchEvents
+		if e.Object == nil {
+			return reconcile.Result{RequeueAfter: RequeueImmediately}, nil
+		}
+		res, err := reconciler.Reconcile(ctx, e.Object.(T))
+		errs = multierr.Append(errs, err)
+		results = append(results, res)
```

Comment:
Why use a multi-result here? You're just going through one object with one result/error combo

Reply:
The handling here was meant to be extendable for multiple reconciliation for multiple objects pushed to the watch channel. Dropping this whole section as it will not be necessary, since I will be handling taking this comment above: https://github.com/awslabs/operatorpkg/pull/124/files#r1930026657